core/trino-main/src/main/java/io/trino/memory/LeastWastedEffortLowMemoryTaskKiller.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.memory;
+
+import com.google.common.collect.ImmutableSet;
+import io.trino.execution.TaskId;
+import io.trino.execution.TaskInfo;
+import io.trino.operator.RetryPolicy;
+import io.trino.operator.TaskStats;
+import io.trino.spi.QueryId;
+import io.trino.spi.memory.MemoryPoolInfo;
+
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static java.util.Comparator.comparing;
+
+public class LeastWastedEffortLowMemoryTaskKiller
```

Comment:
Do we need a unit test covering this?

Reply:
Yeah - will add - meant to do that and forgot.